alf/algorithms/mcts_models.py

```diff
@@ -174,9 +191,15 @@ def calc_loss(self, model_output: ModelOutput,
         loss = loss + policy_loss
         entropy, entropy_for_gradient = dist_utils.entropy_with_fallback(
             model_output.action_distribution)
-        if self._entropy_regularization != 0:
-            loss = loss - self._entropy_regularization * entropy_for_gradient.sum(
+        if self._log_alpha is not None:
+            alpha = self._log_alpha.exp().detach()
+            loss = loss - alpha * (loss_scale * entropy_for_gradient).sum(
                 dim=1)
+            if self._target_entropy is not None:
+                # For some unknown reason, there are memory leaks for not using
+                # detach()
+                self._log_alpha -= self._alpha_adjust_rate * (
+                    entropy.mean() - self._target_entropy()).sign().detach()
```

Comment:
How important is it to use ``sign`` here?

Reply:
I used sign() so the update is invariant to the magnitude of entropy, which is similar to adam.